### DISCUSSION TITLE
Config Global Compatibility Rule

### DISCUSSION THREAD

**Discussion by ikstewa** [Apicurio/apicurio-registry#4752] [Q&A] [answered]
Is it possible to configure the global compatibility rule via env config so it's configured with our deployments?

**Comment by EricWittmann** [comment, marked as answer]
Yes there is, actually.  All of the rules can be configured via ENV vars.  The prefix is `REGISTRY_RULES_GLOBAL_` followed by the rule type, with the value of whatever you want the config to be.  The rule types are:

- `VALIDITY`
- `COMPATIBILITY`
- `INTEGRITY`

Possible values are:

- Compatibility: https://github.com/Apicurio/apicurio-registry/blob/main/schema-util/common/src/main/java/io/apicurio/registry/rules/compatibility/CompatibilityLevel.java#L4
- Validity: https://github.com/Apicurio/apicurio-registry/blob/main/schema-util/common/src/main/java/io/apicurio/registry/rules/validity/ValidityLevel.java#L9
- Integrity: https://github.com/Apicurio/apicurio-registry/blob/main/schema-util/common/src/main/java/io/apicurio/registry/rules/integrity/IntegrityLevel.java#L8

So to set the compatibility rule to `FULL` you would have the following env var:

`REGISTRY_RULES_GLOBAL_COMPATIBILITY=FULL`

---

Note 1:  in Registry 3.0 the prefix changes to `APICURIO_RULES_GLOBAL_`
Note 2:  the value for the integrity rule can be a comma separated list of values - for the other two rules, the values are mutually exclusive
